### ISSUE TITLE
[HOLD for payment 2024-03-26] [HOLD on E/E #347391][$500] [HIGH] for Collect: Show "Unapprove" report action in NewDot

### ISSUE THREAD

**Issue by Beamanator** [Expensify/App#36023]
Related to https://github.com/Expensify/Expensify/issues/364085

# Flow:

## Pre-testing steps

1. Create a new collect policy on OldDot and; 
    1. Enable scheduled submit with a frequency of `Instant` (this should happen by default with Expensify accounts, or accounts on `paidPolicyInstantSubmit` beta)
    1. Make sure the approval mode is "Submit & Approve"
    1. Invite an additional user to the policy to play the member role in these testing flows
1. Run the following script in the chrome console (in OldDot, when the workspace Settings page is open) to enable the policy for workspace creation for a paid plan.
    ```
    p = Policy.getCurrent();
    p.policy.isPolicyExpenseChatEnabled = "true";
    p.save();
    ```

## Test steps

1. In OldDot: As submitter, make sure ^ new workspace is your "Active" / default workspace
2. In OldDot: As submitter, create a report and add an expense
5. In OldDot: As **Admin**, approve the report - then **Unapprove** the report
6. In NewDot: As both accounts, view the expense report & verify you can see both of these report actions:
    1. `<admin> final approved this report`
    2. `<admin> unapproved this report`

Currently we're blocked on 4 ^ - the `<admin> unapproved this report` report action does not appear, even though the report action exists in Onyx
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012cccf392c943b52d</li>
        <li>Upwork Job ID: 1755224934075211776</li>
        <li>Last Price Increase: 2024-02-07</li>
    </ul>
</details>

**Comment by Beamanator** [comment]
Getting some feedback in slack first - internal channel - https://expensify.slack.com/archives/C02NZ2HGJAZ/p1707314650284219

**Comment by rojiphil** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The `UNAPPROVED` action is not currently shown in the collect workspace in NEWDOT when an `APPROVED` report is unapproved.

### What is the root cause of that problem?

[Here](https://github.com/Expensify/App/blob/fe61edaa394533f6ddcfd850a77c6475e6158bbd/src/pages/home/report/ReportActionItemMessage.tsx#L65) we filter the type of actions that needs to be displayed. Currently, we only include `APPROVED` and `SUBMITTED`. Since we have not included `UNAPPROVED` action, this is not displayed. 

### What changes do you think we should make in order to solve the problem?

We can solve this problem by adding another constant [here](https://github.com/Expensify/App/blob/fe61edaa394533f6ddcfd850a77c6475e6158bbd/src/CONST.ts#L535) for `UNAPPROVED` and update the line [here](https://github.com/Expensify/App/blob/fe61edaa394533f6ddcfd850a77c6475e6158bbd/src/pages/home/report/ReportActionItemMessage.tsx#L65) like this to include `UNAPPROVED` action.

    const isApprovedOrSubmittedReportAction = [CONST.REPORT.ACTIONS.TYPE.UNAPPROVED, CONST.REPORT.ACTIONS.TYPE.APPROVED, CONST.REPORT.ACTIONS.TYPE.SUBMITTED].some((type) => type === action.actionName);

We may also want to change the usage of variable `isApprovedOrSubmittedReportAction` to something more generic than specific.

### What alternative solutions did you explore? (Optional)

**Comment by Beamanator** [comment]
still waiting

**Comment by Beamanator** [comment]
Hey @twisterdotcom , any chance you could take a look at this one today? 🙏

**Comment by Beamanator** [comment]
Oh now I see this is on hold for a wave8 issue - maybe we should update the project & make this a `Weekly`?

**Comment by twisterdotcom** [comment]
We will show the oldDot actions for this soon.

**Comment by twisterdotcom** [comment]
Still held.

**Comment by twisterdotcom** [comment]
Okay, gonna use this issue as the main one for the PR. Gonna assign @eh2077 who has offered some help in Slack.

**Comment by twisterdotcom** [comment]
BOOO UPWORK

**Comment by rojiphil** [comment]
@twisterdotcom I am confused. The reason is that I already have a [proposal here](https://github.com/Expensify/App/issues/36023#issuecomment-1932511951) and we were waiting for internal discussions  to complete as mentioned [here](https://github.com/Expensify/App/issues/36023#issuecomment-1933775268) cc @Beamanator

**Comment by twisterdotcom** [comment]
Yup, sorry, we had those discussions and currently landed on the solution in the PR which is slightly different. I was trying to do this myself, but I reached out to a C+ for some help finishing it off.

**Comment by twisterdotcom** [comment]
Nice, added you! Might take up to an hour to show.

**Comment by twisterdotcom** [comment]
Done! Again, could take an hour to show.

**Comment by twisterdotcom** [comment]
Payment Summary:

- @hoangzinh paid $500 [here](https://www.upwork.com/nx/wm/offer/101609537) (C+)
- @eh2077 paid $500 [here](https://www.upwork.com/nx/wm/offer/101609516) (Contributor)

**Comment by eh2077** [comment]
@twisterdotcom I accepted the offer. Can you help to approve the payment please? Thanks.
